File: app/src/main/java/com/water/datasample/helper/FileHelper.java
package com.water.datasample.helper;

import android.content.Context;
import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;

public class FileHelper {
    private Context _ctx;

    public FileHelper(Context ctx) {
        this._ctx = ctx;
    }

    // 向指定的文件中写入指定的数据
    public void save(String filename, String message)
            throws IOException{
        FileOutputStream fout = _ctx.openFileOutput(filename,
                Context.MODE_PRIVATE + Context.MODE_APPEND);//获得FileOutputStream

        //将要写入的字符串转换为byte数组
        byte[] bytes = message.getBytes();
        fout.write(bytes);//将byte数组写入文件
        fout.close();//关闭文件输出流
    }

    /*
     * 这里定义的是文件读取的方法
     * */
    public String read(String filename) throws IOException {
        //打开文件输入流
        FileInputStream fis = _ctx.openFileInput(filename);
        byte[] temp = new byte[1024];
        StringBuilder sb = new StringBuilder("");
        int len = 0;

        //读取文件内容:
        while ((len = fis.read(temp)) > 0) {
            sb.append(new String(temp, 0, len));
        }

        //关闭输入流
        fis.close();
        return sb.toString();
    }

    public boolean delete(String filename) {
        return  _ctx.deleteFile(filename);
    }

    public String[] fileList() {
        return  _ctx.fileList();
    }
}

File: app/src/main/java/com/github/androidhappyclub/datasample/helper/FileHelper.kt
/*
 * MIT License
 *
 * Copyright (c) 2023 AndroidHappyClub
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.github.androidhappyclub.datasample.helper

import android.content.Context
import java.io.IOException

class FileHelper(private val context: Context) {

    /**
     * 向指定的文件中写入指定的数据
     */
    fun save(filename: String, message: String) {
        // 获得FileOutputStream
        val fileOutputStream =
            context.openFileOutput(filename, Context.MODE_PRIVATE + Context.MODE_APPEND)
        val bytes = message.toByteArray() // 将要写入的字符串转换为byte数组
        fileOutputStream.write(bytes) //将byte数组写入文件
        fileOutputStream.close() //关闭文件输出流
    }

    /**
     * 这里定义的是文件读取的方法
     */
    fun read(filename: String): String {
        //打开文件输入流
        val fis = context.openFileInput(filename)
        val temp = ByteArray(1024)
        val sb = StringBuilder()
        var len: Int

        //读取文件内容:
        while (fis.read(temp).also { len = it } > 0) {
            sb.append(String(temp, 0, len))
        }

        //关闭输入流
        fis.close()
        return sb.toString()
    }

    fun delete(filename: String): Boolean {
        return context.deleteFile(filename)
    }

    fun fileList(): Array<String> {
        return context.fileList()
    }
}